packages/website/plugins/generated-rule-docs.ts

```diff
@@ -40,22 +46,211 @@ const generatedRuleDocs: Plugin = () => {
       ) ??
       // ...or at the very end, if not.
       parent.children.length;
 
     // The actual content will be injected on client side.
-    const attrNode = {
+    const attributesNode = {
       type: 'jsx',
       value: `<rule-attributes name="${file.stem}" />`,
     };
-    parent.children.splice(h2Idx, 0, attrNode);
+    parent.children.splice(attributesH2Index, 0, attributesNode);
+
+    // 4. Add a description of how to use / options for the rule
+    const optionLevel = meta.docs.recommended === 'error' ? 'error' : 'warn';
+    if (meta.docs.extendsBaseRule) {
+      const howToUseH2Index = parent.children.findIndex(
+        createH2TextFilter('How to Use'),
+      );
+
+      parent.children.splice(
+        howToUseH2Index + 1,
+        0,
+        {
+          lang: 'jsonc',
+          type: 'code',
+          value: `// .eslintrc.json
+{
+  // note you must disable the base rule as it can report incorrect errors
+  "${file.stem}": "off",
+  "@typescript-eslint/${file.stem}": "${optionLevel}"
+}`,
+        } as mdast.Code,
+        {
+          children: [
+            {
+              type: 'text',
+              value: 'Options',
+            },
+          ],
+          depth: 2,
+          type: 'heading',
+        } as mdast.Heading,
+        {
+          children: [
+            {
+              value: 'See ',
+              type: 'text',
+            },
+            {
+              children: [
+                {
+                  type: 'inlineCode',
+                  value: `eslint/${file.stem}`,
+                },
+                {
+                  type: 'text',
+                  value: ' options',
+                },
+              ],
+              type: 'link',
+              url: `https://eslint.org/docs/rules/${file.stem}#options`,
+            },
+            {
+              type: 'text',
+              value: '.',
+            },
+          ],
+          type: 'paragraph',
+        } as mdast.Paragraph,
+        // todo: add more?
+      );
+    } else {
+      let optionsH2Index = parent.children.findIndex(
+        createH2TextFilter('Options'),
+      );
+
+      if (optionsH2Index === -1) {
+        const whenNotToUseItH2Index = parent.children.findIndex(
+          createH2TextFilter('When Not To Use It'),
+        );
+        optionsH2Index =
+          whenNotToUseItH2Index === -1
+            ? parent.children.length - 1
+            : whenNotToUseItH2Index;
+
+        parent.children.splice(optionsH2Index, 0, {
+          children: [
+            {
+              type: 'text',
+              value: 'Options',
+            },
+          ],
+          depth: 2,
+          type: 'heading',
+        } as mdast.Heading);
+      }
+
+      if (meta.schema.length === 0) {
+        parent.children.splice(
+          optionsH2Index + 1,
+          0,
+          {
+            lang: 'jsonc',
+            type: 'code',
+            value: `// .eslintrc.json
```

Comment:
Should we generate these as `.cjs` to match the recommendations in our other docs?